File: src/main/java/ir/mmd/intellijDev/Actionable/lang/html/settings/SettingsState.java
package ir.mmd.intellijDev.Actionable.lang.html.settings;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import ir.mmd.intellijDev.Actionable.lang.html.type.ExpandTagOnType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@State(
	name = "ir.mmd.intellijDev.Actionable.lang.html.settings.SettingsState",
	storages = @Storage("Actionable.Lang.HTML.SettingsState.xml")
)
public class SettingsState implements PersistentStateComponent<SettingsState> {
	/**
	 * @see ExpandTagOnType
	 */
	public boolean expandTagOnTypeEnabled = false;
	
	@Override
	public @Nullable SettingsState getState() {
		return this;
	}
	
	@Override
	public void loadState(@NotNull SettingsState state) {
		XmlSerializerUtil.copyBean(state, this);
	}
}

File: src/main/java/ir/mmd/intellijDev/Actionable/lang/html/settings/SettingsState.kt
package ir.mmd.intellijDev.Actionable.lang.html.settings

import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage
import com.intellij.util.xmlb.XmlSerializerUtil

@State(
	name = "ir.mmd.intellijDev.Actionable.lang.html.settings.SettingsState",
	storages = [Storage("Actionable.Lang.HTML.SettingsState.xml")]
)
class SettingsState : PersistentStateComponent<SettingsState?> {
	/**
	 * @see ir.mmd.intellijDev.Actionable.lang.html.type.ExpandTagOnType
	 */
	var expandTagOnTypeEnabled: Boolean = false
	
	override fun getState() = this
	override fun loadState(state: SettingsState) = XmlSerializerUtil.copyBean(state, this)
}
